infer/environment.cc

```diff
@@ -928,6 +928,7 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
         core::TypeAndOrigins tp;
         bool noLoopChecking = cfg::isa_instruction<cfg::Alias>(bind.value.get()) ||
                               cfg::isa_instruction<cfg::LoadArg>(bind.value.get()) ||
+                              bind.bind.variable == cfg::LocalRef::selfVariable() ||
                               cfg::isa_instruction<cfg::LoadSelf>(bind.value.get());
```

Comment:
I am pretty sure this line is redundant with the `variable == selfVariable` check on the line above. Did you add it back because you found that it was not in fact redundant?